tool/src/org/antlr/v4/tool/Grammar.java

```diff
@@ -430,6 +430,10 @@ public void defineAction(GrammarAST atAST) {
         }
     }
 
+	public void removeParserHeader() {
+		namedActions.remove("header");
+	}
```

Comment:
I'm against adding such method to `Grammar`. Grammar should not have such responsibility and it looks like workaround.

Reply:
Grammar should definitely have responsibility for managing its members. That's basic OOP.

Reply:
It should but related methods. It's unclear the need of method of removing `header` named action.